src/treelearner/feature_histogram.hpp

```diff
@@ -94,16 +95,20 @@ class FeatureHistogram {
     double gain_shift = GetLeafSplitGain(sum_gradient, sum_hessian,
                                          meta_->config->lambda_l1, meta_->config->lambda_l2, meta_->config->max_delta_step);
     double min_gain_shift = gain_shift + meta_->config->min_gain_to_split;
+    int rand_threshold = 0;
+    if (meta_->num_bin - 2 > 0){
+      rand_threshold = rand_.NextInt(0, std::max(meta_->num_bin - 2, 0));
+    }
```

Comment:
Seems that `std::max(meta_->num_bin - 2, 0)` is excess here because `meta_->num_bin - 2 > 0`.